src/fosslight_dependency/_package_manager.py

```
@@ -87,122 +88,203 @@ def parse_direct_dependencies(self):
+
+                if self.package_manager_name in (const.ANDROID, const.KOTLIN):
+                    if module_gradle_backup and os.path.isfile(module_gradle_backup):
+                        if os.path.isfile(module_build_gradle):
+                            os.remove(module_build_gradle)
+                        shutil.move(module_gradle_backup, module_build_gradle)
+
+            if os.path.isfile(self.input_file_name):
+                logger.info(f'Found {self.input_file_name}, skip to run plugin.')
+                self.set_direct_dependencies(False)
+                ret_task = True
+        except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)